src/compiler/types.ts

```diff
@@ -2141,18 +2148,23 @@ export interface KeywordTypeNode<TKind extends KeywordTypeSyntaxKind = KeywordTy
     readonly kind: TKind;
 }
 
-export interface ImportTypeAssertionContainer extends Node {
-    readonly kind: SyntaxKind.ImportTypeAssertionContainer;
+/** @deprecated */
+export type ImportTypeAssertionContainer = ImportTypeAttributes;
+
+export interface ImportTypeAttributes extends Node {
+    readonly kind: SyntaxKind.ImportTypeAttributes;
     readonly parent: ImportTypeNode;
-    readonly assertClause: AssertClause;
+    /** @deprecated */ readonly assertClause: AssertClause;
+    readonly attributes: ImportAttributes;
     readonly multiLine?: boolean;
 }
 

```

Comment:
This interface seems useless - but it already exists, so I guess we're only keeping it for backwards compatibility?

Reply:
It was probably made like this to "future-proof" things in case we needed more than just import assertions - but now that they're called "attributes" it probably makes less sense.

Reply:
Maybe `ImportTypeNode["attributes"]` could skip this interface and just have the type `ImportAttributes`.

Reply:
Yeah, I think the new `attributes` API can diverge and not use it, if that doesn’t cause big problems elsewhere. So e.g.

```ts
importTypeNode.assertions.assertClause === importTypeNode.attributes
```